File: narcissistic.rb
def narc(number)
  power = number.to_s.length
  array = number.to_s.chars
  answer = 0
  array.each do |number|
    answer += number.to_i**power
  end
  return answer == number
end

p narc(153) #=> true
p narc(1634) #=> true
p narc(28) #=> false

File: javascript/narcissistic.js
function narc(number) {
  var power = number.toString().length;
  var array = number.toString().split('');
  answer = 0;
  array.forEach(function(number) {
    answer += Math.pow(number, power)
  })
  return answer === number;
}

console.log(narc(153)) //=> true
console.log(narc(1634)) //=> true
console.log(narc(28)) //=> false
